src/webots/gui/WbMainWindow.cpp

```diff
@@ -1747,6 +1747,21 @@ void WbMainWindow::uploadStatus() {
     WbMessageBox::critical(tr("Upload failed: Upload status could not be modified."));
 }
 
+bool WbMainWindow::uploadFileExists(QString filename) {
+  int maxIterations = 10;
+  while (!QFileInfo(WbStandardPaths::webotsTmpPath() + filename).exists() && maxIterations) {
+    QTime delay = QTime::currentTime().addMSecs(100);
+    while (QTime::currentTime() < delay)
+      QCoreApplication::processEvents(QEventLoop::AllEvents);
```

Comment:
Calling `QCoreApplication::processEvents` in an active loop is probably super computer intensive. I believe we should avoid this:
```suggestion
    QCoreApplication::processEvents(QEventLoop::AllEvents, 100);
```

Reply:
I have changed this for a QThread::msleep(100) instead.